### ISSUE TITLE
Tagging recipes

### ISSUE THREAD

**Issue by pscn** [Korlimann/MixnJuice#8]
> The option to “tag” things. For example, with the tag “strawberry”, users can not only search for recipe names which include the word “strawberry” but rather for all recipes including the tag “strawberry”. Although, coding-wise, I can imagine it would be easier to simply search the recipe, if it includes a strawberry. This way, users won’t have to add a tag for each flavour they used. Tags could still be implemented for other stuff like “shake&vape”, “easy&quick mix” or “dtl”, and so on.

It would be cool if the system was smart enough to suggest tags for recipes based on:

- tags of the flavor, provided we use tags for flavors
- tags commonly used for recipes with this flavor

My fear with tags is that it can generate a lot of noise if they can be added too easily.  We should encourage the user to use existing tags before adding a new one.

**Comment by chommsi** [comment]
Yes, we should definitely encourage the usage of existing tags. Primarily by suggesting up-front, in a way like "We see you used this flavours: [flavours] Would you like to add the tags for the flavours to your recipe? [yes] [no]"
We could also try and implement something that starts to auto-complete when a user types, for example you type "straw" and get suggested "strawberry", "strawberry juice", "strawberry jam", etc.

Also, I feel like this is one of the most wanted features from the community, so I'd say this is one of the first things we should tackle.
